source/common/network/address_impl.cc

```diff
@@ -280,6 +280,7 @@ int PipeInstance::bind(int fd) const {
     return ::bind(fd, reinterpret_cast<const sockaddr*>(&address_),
                   offsetof(struct sockaddr_un, sun_path) + address_length_);
   }
+  unlink(address_.sun_path);
```

Comment:
Should we `RELEASE_ASSERT` that this is successful? Is this behavior tested?

Reply:
Added a test. If this fails, the bind will return an appropriate errno depending on whether the existing file is a pre-existing socket, or some other file type.

Reply:
Maybe just add a comment here. I generally find that when reading code, if an error is deliberately ignored it's helpful to know this vs. suspecting some missing error handling.

Reply:
Done.